### ISSUE TITLE
Cluster creation hanging for >20m

### ISSUE THREAD

**Issue by zmerlynn** [kubernetes/kops#866]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws/1001
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws/1002
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws/1003
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws/1010
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-kops-aws/1015

This is the most frequent failure right now. It's not clear what's causing these, or where to start with diagnosis.

**Comment by chrislovecnm** [comment]
How can I re-run this tests??

**Comment by zmerlynn** [comment]
From the k8s repo, and $KOPS pointing to a kops binary, $CLUSTER being your desired e2e cluster FQDN, $STATE_STORE being your state store, and `v1.6.0-alpha.0.351+a330acddee9f66` is the version for 1001-1003 (but it did succeed on another run with that version, incidentally):

```
go run hack/e2e.go -deployment kops -v -up -down -kops $KOPS -kops-cluster $CLUSTER -kops-state 
$STATE_STORE --kops-kubernetes-version https://storage.googleapis.com/kubernetes-release-dev/ci/v1.6.0-alpha.0.351+a330acddee9f66
```

**Comment by zmerlynn** [comment]
That *just* turns a cluster up and back down. Ping if you want to see how to run tests, but it's basically the `-test` option with the right options. But this problem is just up/down nastiness.

**Comment by chrislovecnm** [comment]
@zmerlynn we are having issues that have problems with the 1.5 release.  Can you get on to a master or a node and get us some logs?

**Comment by zmerlynn** [comment]
@chrislovecnm: The Jenkins e2e infrastructure does have a method to capture logs before teardown, but right now the way it's handled for `kops` relies on `kubectl get nodes` working. A local repro could obviously do something different, though.

**Comment by chrislovecnm** [comment]
Zack I think a how to track this info down 101. Can we sechule a time to have you walk me and @kris-nova through this??

**Comment by yissachar** [comment]
@zmerlynn I'm wondering if this is related to https://github.com/kubernetes/kops/issues/859. Seems like it could the same issue based on kubectl not being able to hit a fresh cluster, and a user reporting that it took ~20 minutes for their cluster to be fully operational.

**Comment by justinsb** [comment]
This is my top priority now that we're blocking PRs - well played @zmerlynn  ;-)   Seriously though I am very grateful to you for getting that up and running and am going to try out a few ideas this morning!

Once we get it working I think we should have the PR builder switch to the stable kops branch (but only once we fix these timeouts)

**Comment by chrislovecnm** [comment]
@zmerlynn are you running master, a release or a sha?  Just wondering if we should cut 1.4.2 or 1.4.6 for this ...

**Comment by chrislovecnm** [comment]
@justinsb / @zmerlynn where are we at with this??

**Comment by zmerlynn** [comment]
@chrislovecnm: Latest CI `kops` deploying latest CI `kubernetes`. Yes, it's likely #859 - whenever I've had trouble with it locally, I think it's always been DNS related.

**Comment by zmerlynn** [comment]
@justinsb: How is the stable branch managed? I'd rather have an automated solution where the [`kops` job](https://k8s-testgrid.appspot.com/google-aws#kops-aws) just updates a `kops` `latest-green.txt`, then the PR builder can use that. That gets us the tip of both, but the PR builder always gets a `kops` build that was known to pass tests.

**Comment by chrislovecnm** [comment]
So this is all DNS based issues?

**Comment by zmerlynn** [comment]
I think this will be fixed by #1006. If it's not, will reopen.
